lxd/container.go

```diff
@@ -384,17 +384,17 @@ func instanceCreateAsSnapshot(s *state.State, args db.InstanceArgs, sourceInstan
 		 * after snapshotting will fail.
 		 */
 
-		criuMigrationArgs := CriuMigrationArgs{
-			cmd:          lxc.MIGRATE_DUMP,
-			stateDir:     stateDir,
-			function:     "snapshot",
-			stop:         false,
-			actionScript: false,
-			dumpDir:      "",
-			preDumpDir:   "",
+		criuMigrationArgs := instance.CriuMigrationArgs{
+			Cmd:          lxc.MIGRATE_DUMP,
+			StateDir:     stateDir,
+			Function:     "snapshot",
+			Stop:         false,
+			ActionScript: false,
+			DumpDir:      "",
+			PreDumpDir:   "",
 		}
 
-		c := sourceInstance.(*containerLXC)
+		c := sourceInstance.(instance.Container)
```

Comment:
I guess this one will sort itself out when we figure out Migrate() for VMs?

Reply:
@stgraber indeed, I could move this to `Instance` now and just return the VM one as not implemented for now.

Reply:
Yeah, might as well, we do want VMs to have a Migrate function, we'll move to a different signature though (not sure what yet) so it applies to both containers and VMs, but adding it to the interface as it is now is fine.